### PR TITLE
Deprecate `serialiseTxLedgerCddl`

### PR DESCRIPTION
**Pull request by palas**
# Changelog

```yaml
- description: |
    Update `cardano-cli` and tests to work with the changes to deprecate `serialiseTxLedgerCddl`
  type:
    - breaking       # the API has changed in a breaking way
```

# Context

## `cardano-api`

- Main PR: https://github.com/IntersectMBO/cardano-api/pull/534
- Branch: [cardano-api/deprecate-serialiseTxLedgerCddl](https://github.com/IntersectMBO/cardano-api/tree/deprecate-serialiseTxLedgerCddl)
- Backported branch: [cardano-api/deprecate-serialiseTxLedgerCddl-testing](https://github.com/IntersectMBO/cardano-api/tree/deprecate-serialiseTxLedgerCddl-testing)

## `cardano-cli`

- PR: https://github.com/IntersectMBO/cardano-cli/pull/772
- In the end, done in PR: https://github.com/IntersectMBO/cardano-cli/pull/786
- Branch: [cardano-cli/deprecate-serialiseTxLedgerCddl-testing](https://github.com/IntersectMBO/cardano-cli/tree/deprecate-serialiseTxLedgerCddl-testing)

## `cardano-node`

- PR: https://github.com/IntersectMBO/cardano-node/pull/5867
- In the end, done in PR: https://github.com/IntersectMBO/cardano-node/pull/5879
- Branch: [cardano-node/deprecate-serialiseTxLedgerCddl-testing](https://github.com/IntersectMBO/cardano-node/tree/deprecate-serialiseTxLedgerCddl-testing)

# How to trust this PR

Don't trust it. Make sure the changes make sense and that the breakages in compatibility are for the better.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/Read.hs [hunk 1]
```diff
@@ -632,16 +628,16 @@ acceptKeyWitnessCDDLSerialisation err =
...
 readCddlWitness fp = do
   readFileTextEnvelopeCddlAnyOf teTypes fp
  where
-  teTypes = [ FromCDDLWitness "TxWitness ShelleyEra" CddlWitness
-            , FromCDDLWitness "TxWitness AllegraEra" CddlWitness
-            , FromCDDLWitness "TxWitness MaryEra" CddlWitness
-            , FromCDDLWitness "TxWitness AlonzoEra" CddlWitness
-            , FromCDDLWitness "TxWitness BabbageEra" CddlWitness
-            , FromCDDLWitness "TxWitness ConwayEra" CddlWitness
+  teTypes = [ FromCDDLWitness "TxWitness ShelleyEra" coerce
```
Comment (carbolymer) [inline cardano-cli/src/Cardano/CLI/Read.hs]: Why do we need `coerce` here? Isn't that `InAnyShelleyBasedEra KeyWitness -> InAnyShelleyBasedEra KeyWitness`, so `id`?

Comment (palas) [reply cardano-cli/src/Cardano/CLI/Read.hs]: It seems that it is no longer necessary in my new approach. I am not sure how that worked, TBH, it was trial and error...

**Review by smelc — APPROVED** [review]

**Comment by smelc** [comment]
Integrated into https://github.com/IntersectMBO/cardano-cli/pull/786

**Comment by smelc** [comment]
Closing as it got merged as part of https://github.com/IntersectMBO/cardano-cli/pull/786

**Pull request closed without merging**